client/PlayerConnectionThreadsBuilder.py

```diff
@@ -0,0 +1,103 @@
+from PlayerConnectionConfig import PlayerConnectionConfig
+from ThreadStatus import ThreadStatus
+from ConnectionThread import ConnectionThread
+from PlayerConnectionThreadHandler import PlayerConnectionThreadHandler
+
+
+class PlayerConnectionThreadsBuilder:
+    def __init__(self):
+        self.username = None
+        self.users = None
+        self.ports = None
+        self.client_index = None
+        self.client_configs = None
+        self.other_players_configs = None
+        self.number_of_active_connections = None
+        self.client_to_ports_config = None
+
+    def convert_server_response_to_users_and_ports(self, response):
+        '''Split response to users list and ports list'''
+        self.users = response[:-1]
+        self.ports = response[-1]
+
+    def find_client_index_in_users(self, username):
+        '''Find current client's index in users list'''
+        self.username = username
+        if self.users is not None:
+            for i in range(len(self.users)):
+                if self.users[i][0] == username:
+                    self.client_index = i
+                    return
+        else:
+            print('self.users is needed!')
+
+    def prepare_connection_config_for_each_player(self):
+        '''Create config for each player with their IP and ports'''
+        if self.username is not None and \
+                self.users is not None and \
+                self.ports is not None:
+            self.client_configs = []
+            self.other_players_configs = []
+            for i in range(len(self.users)):
+                user_ports = self.ports[i * (len(self.users)):(i + 1) * (len(self.users))]
+                player_config = PlayerConnectionConfig(self.users[i][0], self.users[i][1], user_ports)
+                self.client_configs.append(player_config)
+                if self.users[i][0] != self.username:
+                    self.other_players_configs.append(player_config)
+        else:
+            print('username, users and ports are needed!')
+
+    def get_number_of_active_connections(self):
+        '''Get how many connections should be created actively(client connects, rather than listens)'''
+        if self.users is not None and \
+                self.client_index is not None:
```

Comment:
`if None not in [self.users, self.client_index]`

Reply:
True, that's cleaner.